lib/tdf3/src/tdf.ts

```diff
@@ -951,60 +952,129 @@ export class TDF extends EventEmitter {
     // TODO: Don't await on each segment serially, instead use event-driven approach to prevent deadlock.
     // See: https://github.com/jherwitz/tdf3-js/blob/3ec3c8a3b8c5cecb6f6976b540d5ecde21183c8c/src/tdf.js#L739
     let encryptedOffset = 0;
+    let pullController: ReadableStreamDefaultController;
 
-    // eslint-disable-next-line @typescript-eslint/no-this-alias
-    // const that = this;
+    const asyncQueueWorker = async (decryptedChunk: any) => {
+      return pullController.enqueue(await decryptedChunk());
+    };
+
+    const q: any = fastq.promise(asyncQueueWorker, 1);
 
     const underlyingSource = {
       pull: async (controller: ReadableStreamDefaultController) => {
-        if (segments.length === 0) {
-          controller.close();
-          return;
-        }
+        pullController = controller;
+        // @ts-ignore
+        if (rcaParams && rcaParams.wu) {
```

Comment:
This should be a new option, not recycling `.wu`. AFAICt this is independent of the chunker type / source. e.g. it should work on any chunk able source, right?

```suggestion
        if (concurrentReadsEnabled) {
```

and then add `concurrentReadsEnabled` to the method parameters